src/analyze/flavors/lit-element/parse-lit-property-configuration.ts

```diff
@@ -61,124 +61,124 @@ export function getLitElementPropertyDecoratorConfig(node: Node, context: Analyz
+ */
+function hasProperty<T extends string>(obj: object, key: T): obj is { [K in T]: unknown } {
+	return Object.prototype.hasOwnProperty.call(obj, key);
+}
+
+/**
+ * Computes the correct type for a given node for use in lit property
+ * configuration.
+ * @param ts
+ * @param node
+ */
+export function getLitPropertyType(ts: typeof tsModule, node: Node): SimpleType | string {
+	const value = ts.isIdentifier(node) ? node.text : undefined;
+
+	switch (value) {
+		case "String":
+		case "StringConstructor":
+			return { kind: "STRING" };
+		case "Number":
+		case "NumberConstructor":
+			return { kind: "NUMBER" };
+		case "Boolean":
+		case "BooleanConstructor":
+			return { kind: "BOOLEAN" };
+		case "Array":
+		case "ArrayConstructor":
+			return { kind: "ARRAY", type: { kind: "ANY" } };
+		case "Object":
+		case "ObjectConstructor":
+			return { kind: "OBJECT", members: [] };
+		default:
+			// This is an unknown type, so set the name as a string
+			return node.getText();
+	}
+}
+
 /**
  * Parses an object literal expression and returns a lit property configuration.
  * @param node
  * @param existingConfig
  * @param context
  */
 export function getLitPropertyOptions(
-	node: ObjectLiteralExpression,
+	node: Node,
+	object: unknown,
 	context: AnalyzerVisitContext,
 	existingConfig: LitElementPropertyConfig = {}
 ): LitElementPropertyConfig {
 	const { ts } = context;
+	const result: LitElementPropertyConfig = { ...existingConfig };
+	let attributeInitializer: Node | undefined;
+	let typeInitializer: Node | undefined;
 
-	// Build up the property configuration by looking at properties in the object literal expression
-	return node.properties.reduce((config, property) => {
-		if (!ts.isPropertyAssignment(property)) return config;
-
-		const initializer = property.initializer;
-		const kind = ts.isIdentifier(property.name) ? property.name.text : undefined;
-
-		return parseLitPropertyOption({ kind, initializer, config }, context);
-	}, existingConfig);
-}
-
-export function parseLitPropertyOption(
-	{ kind, initializer, config }: { kind: string | undefined; initializer: Expression; config: LitElementPropertyConfig },
-	context: AnalyzerVisitContext
-): LitElementPropertyConfig {
-	const { ts, checker } = context;
-
-	// noinspection DuplicateCaseLabelJS
-	switch (kind) {
-		case "converter": {
-			return { ...config, hasConverter: true };
+	if (typeof object === "object" && object !== null && !Array.isArray(object)) {
+		if (hasProperty(object, "converter") && object.converter !== undefined) {
+			result.hasConverter = true;
 		}
 
-		case "reflect": {
-			return { ...config, reflect: resolveNodeValue(initializer, context)?.value === true };
+		if (hasProperty(object, "reflect") && object.reflect !== undefined) {
+			result.reflect = object.reflect === true;
 		}
 
-		case "state": {
-			return { ...config, state: resolveNodeValue(initializer, context)?.value === true };
+		if (hasProperty(object, "state") && object.state !== undefined) {
```

Comment:
huh, I wonder what this is for

Reply:
ah, this one is documented at https://lit.dev/docs/components/properties/#property-options I was looking on the wrong page